v2/internal/pkg/operator/filtered_collector.go

```diff
@@ -0,0 +1,310 @@
+package operator
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"maps"
+	"os"
+	"path"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"github.com/opencontainers/go-digest"
+	"github.com/openshift/oc-mirror/v2/internal/pkg/api/v2alpha1"
+	"github.com/openshift/oc-mirror/v2/internal/pkg/image"
+	"github.com/openshift/oc-mirror/v2/internal/pkg/mirror"
+	"github.com/otiai10/copy"
+)
+
+type FilterCollector struct {
+	OperatorCollector
+}
+
+// OperatorImageCollector - this looks into the operator index image
+// taking into account the mode we are in (mirrorToDisk, diskToMirror)
+// the image is downloaded (oci format) and the index.json is inspected
+// once unmarshalled, the links to manifests are inspected
+func (o *FilterCollector) OperatorImageCollector(ctx context.Context) (v2alpha1.CollectorSchema, error) {
```

Comment:
@sherine-k this block is duplicated here `internal/pkg/operator/local_stored_collector.go` - is the any way to refactor into one function for maintainability

Reply:
This is true. We need to refactor at some point in time